ts/kpt-functions/src/types.ts

```diff
@@ -166,6 +166,29 @@ export class Configs {
     return this.functionConfig;
   }
 
+  /**
+   * Returns the functionConfig data field if defined.
+   */
+  getFunctionConfigDataOrThrow() {
+    const cm = this.getFunctionConfig();
+    if (!cm) {
+      throw new FunctionConfigError(
+        `functionConfig expected, instead undefined`
+      );
+    }
+    if (!isConfigMap(cm)) {
+      throw new FunctionConfigError(
+        `functionConfig expected to be of kind ConfigMap, instead got: ${cm.kind}`
+      );
+    }
+    if (!cm.data) {
```

Comment:
cm not having 'data' is a valid state. The caller can then check and throw if specific fields are missing from the map.

Reply:
Done